util/inc/leatherman/util/scoped_descriptor.hpp

```diff
@@ -0,0 +1,27 @@
+/**
```

Comment:
This only works on POSIX. I'm not too tied to putting it in a posix subdirectory, but @peterhuene may have thoughts on that.